### ISSUE TITLE
Accessibility issues for the Tag/Endpoint Sections

### ISSUE THREAD

**Issue by kayla-beckham** [scalar/scalar#3440]
### What happens?

![Screen Shot 2024-10-10 at 9 22 08 AM](https://github.com/user-attachments/assets/e835e938-da5b-4a4d-a917-751c76994a81)

1. When using keyboard navigation over headings that are clickable, there is no visual indication that clicking them will copy something to the clipboard (when you hover over the heading with a mouse, a # appears)
    1. Screen reader announces ‘number <heading name>’ and screen reader will say that there is a clickable copy component, but keyboard users who do not use screen reader have no indication of the copy button.
    2. When a user clicks a copy button with mouse or keyboard, there is no announcement by a screen reader of a status message that the content was copied
2. There is no visual indication that the Endpoint sections in the section container are clickable links.
    1. The links in the Endpoint card at the top of a section do not receive keyboard focus
3. When you move to a new section by keyboard navigation and activate the ‘Show more’ button, the focus should move to the first interactive element in the expanded section

![Screen Shot 2024-10-10 at 9 19 50 AM](https://github.com/user-attachments/assets/f7bd415f-8a48-46b6-af5f-e65e0b7ec27d)

4. On the request code snippet example, the screenreader announcement for the client library combobox is not descriptive. It announces the name of the currently selected library.
    1. There is not visible focus when the keyboard navigates to the combobox of client libraries on code snippets
    2. The verb & path label of the code snippet is not announced by a screenreader
    3. The copy button in the code snippet does not receive visible focus during keyboard navigation. It appears when the mouse hovers over the code snippet
    4. The test request button does not receive visible focus during keyboard navigation.
5. Scrollable code snippets are not keyboard accessible. The Code sections do not receive keyboard focus.
6. In Responses, the toggle arrow icon is missing the following attributes: focusable="false”, role="presentation”, aria-hidden="true"
7. In responses, the element that expands to show response definition for each status code is missing the following attributes: role="button" to be focusable and identified as interactive elements for assistive technologies, Name to be announced by screen reader (Note: the button should have name and be associated with the visual label 'Responses’), state to be announced by screen reader (expanded or collapsed)
8. In request or response sections, when there is nested information with a ‘Show child attributes’ button, the button should have an aria-control attribute that refers to the region that is hidden under the button.
    1. The region hidden under the button should have role=region and an accessible name that is announced by the screen reader
    2. The elements that are revealed after the ‘Show child attributes’ button is pressed should be included in a structured list.
9. When screen is zoomed in very far (for example 400%), the code snippets and response body examples are not scrollable.

### What did you expect to happen?

Buttons that appear on hover should appear when keyboard focus moves to them.
Copy actions should have status messages announced by screen reader.
Active elements like clickable links should have a visual indication that they are clickable and should be able to receive keyboard focus.
The code snippets should be accessible by keyboard and screen reader. They should be announced by a screen reader, and the buttons on the component should be keyboard accessible with contextual screen reader announcements.
Expanding components and regions in the request and response documentation should have correct aria labels.
When you click a button that opens an expandable component, the keyboard focus should move into that component.

**Comment by hwkr** [comment]
Fixed in #3621
